integration/send-files/e2e/utils.ts

```diff
@@ -0,0 +1,47 @@
+import { INestApplication } from '@nestjs/common';
+import { IncomingMessage, request, RequestOptions } from 'http';
+import { URL } from 'url';
+
+export const getHttpBaseOptions = async (
+  app: INestApplication,
+): Promise<URL> => {
+  const url = await app.getUrl();
+  return new URL(url);
+};
+
+export const sendCanceledHttpRequest = async (url: URL) => {
+  return new Promise((resolve, reject) => {
+    const req = request(url, res => {
+      // close the request once we get the first response of data
+      res.on('data', () => {
+        req.destroy();
+      });
+      // response is closed, move on to next request and verify it's doable
+      res.on('close', resolve);
+    });
+    // fire the request
+    req.end();
+  });
+};
+
+export const sendHttpRequest = async (url: URL) => {
+  return new Promise<IncomingMessage>((resolve, reject) => {
+    const req = request(url, res => {
+      // this makes sure that the response actually starts and is read. We could verify this value against the same
+      // that is in an earlier test, but all we care about in _this_ test is that the status code is 200
+      res.on('data', chunk => {
+        /* no op */
```

Comment:
```suggestion
        // no op 
```
Better to use the single comment here as well, as it is used throughout the file

Reply:
Force of habit, I've always written no op comments as `/* no op */`. Shouldn't matter either way but I've made it an inline comment now

Reply:
no problem 😉